test/controllers/HealthReasonControllerSpec.scala

```diff
@@ -278,6 +279,135 @@ class HealthReasonControllerSpec extends SpecBase {
         }
       }
     }
-  }
 
+    "onPageLoadForWhenDidHospitalStayBegin" when {
+
+      "the user is authorised" must {
+
+        "return OK and correct view" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(userRequestWithCorrectKeys)
+          status(result) shouldBe OK
+        }
+
+        "return OK and correct view (pre-populated date when present in session)" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result = controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(fakeRequestConverter(fakeRequestWithCorrectKeys
+            .withSession(SessionKeys.whenHealthIssueStarted -> "2021-01-01")))
+          status(result) shouldBe OK
+          val documentParsed = Jsoup.parse(contentAsString(result))
+          documentParsed.select(".govuk-date-input__input").get(0).attr("value") shouldBe "1"
+          documentParsed.select(".govuk-date-input__input").get(1).attr("value") shouldBe "1"
+          documentParsed.select(".govuk-date-input__input").get(2).attr("value") shouldBe "2021"
+        }
+
+        "user does not have the correct session keys" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)
+          status(result) shouldBe INTERNAL_SERVER_ERROR
+        }
+      }
+
+      "the user is unauthorised" when {
+        "return 403 (FORBIDDEN) when user has no enrolments" in new Setup(AuthTestModels.failedAuthResultNoEnrolments) {
+          val result: Future[Result] = controller.onSubmitForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)
+          status(result) shouldBe FORBIDDEN
+        }
+
+        "return 303 (SEE_OTHER) when user can not be authorised" in new Setup(AuthTestModels.failedAuthResultUnauthorised) {
+          val result: Future[Result] = controller.onSubmitForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)
```

Comment:
Sorry to be a pain! But could you change this to controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)